types/notp/index.d.ts

```diff
@@ -0,0 +1,123 @@
+     *
+     * E.g. if W = 100, and C = 5, this function will check the passcode against
+     * all One Time Passcodes between 5 and 105.
+     *
+     * Default - 50
+     */
+    window?: number;
+
+    /**
+     * Counter value. This should be stored by the application, must be user
+     * specific, and be incremented for each request.
+     */
+    counter?: number;
+}
+
+/**
+ * time?: number
+ * _t?: number
+ */
+export interface TOTPOptions {
+    /**
+     * The time step of the counter. This must be the same for every request and is
+     * used to calculat C.
+     *
+     * Default - 30
+     */
+    time?: number;
+
+    /**
+     * UNIX Epoch time (overwrite time in test environment, NODE_ENV=test)
+     */
+    _t?: number;
+}
+
+/**
+ * delta: number
+ */
+export interface VerifyResult {
+    /**
+     * Time step difference between the client and the server.
+     */
+    delta: number;
+}
+
+/**
+ * hotp.gen(key: string | Buffer | Uint8Array, opt?: HOTPGenerateOptions): string
+ * hotp.verify(token: string, key: string | Buffer | Uint8Array, opt?: HOTPVerifyOptions): VerifyResult | null
+ */
+export namespace hotp {
+    /**
+     * Generate a counter based One Time Password.
+     * @param key Key for the one time password. This should be unique and secret for
+     * every user as this is the seed that is used to calculate the HMAC.
+     * @param opt Generate options.
+     */
+    function gen(key: string | Buffer | Uint8Array, opt?: HOTPGenerateOptions): string;
+
+    /**
+     * Check a One Time Password based on a counter.
+     * @param token Passcode to validate.
+     * @param key Key for the one time password. This should be unique and secret for
+     * every user as it is the seed used to calculate the HMAC.
+     * @param opt Verify options.
+     */
+     function verify(token: string,
+                     key: string | Buffer | Uint8Array,
+                     opt?: HOTPVerifyOptions): VerifyResult | null;
+}
+
+/**
+ * totp.gen(key: string | Buffer | Uint8Array, opt?: TOTPOptions): string
+ * totp.verify(token: string, key: string | Buffer | Uint8Array, opt?: TOTPOptions): VerifyResult | null
+ */
+export namespace totp {
+    /**
+     * Generate a time based One Time Password.
+     * @param key Key for the one time password. This should be unique and secret
+     * for every user as it is the seed used to calculate the HMAC.
+     * @param opt Generate options.
+     *
+     */
+    function gen(key: string | Buffer | Uint8Array, opt?: TOTPOptions): string;
+
+    /**
+     * Check a One Time Password based on a timer.
+     * @param token Passcode to validate.
+     * @param key Key for the one time password. This should be unique and secret
+     * @param opt Verify options.
+     *
+     */
+    function verify(token: string,
+                    key: string | Buffer | Uint8Array,
+                    opt?: TOTPOptions): VerifyResult | null;
```

Comment:
It looks like the totp.verify options also take in an optional value for `window`. Something like:

```
interface TOTPGenerateOptions { time?: number; _t?: number }
interface HOTPGenerateOptions { counter?: number; }
type TOTPVerifyOptions = TOTPGenerateOptions & { window?: number; }
type HOTPVerifyOptions =  HOTPGenerateOptions & { window?: number }
```

Reply:
Hi Pranav, I have made the necessary amendments.